django/src/rdwatch/views/model_run.py

```diff
@@ -379,7 +396,10 @@ def get_model_run(request: HttpRequest, id: UUID4):
 @router.post(
     '/{model_run_id}/site-model/',
     response={201: UUID4},
+    auth=[ModelRunAuth()],
```

Comment:
For some reason it didn't add my other comment.  I use this endpoint in the client for drawing new proposal sites.  So I think we need multiple authentication here:
https://django-ninja.dev/guides/authentication/?h=multiple+auth#multiple-authenticators
That way it can try using the django_auth through gitlab first and then fallback to the API header.

Reply:
To do this, we need to make the CSRF exemption conditional on whether or not the API key auth or the default Django session auth is used. Since disabling CSRF protection for the default auth flow is insecure. I think we can use a custom decorator to do this, I'll look into it.

Reply:
Deferring this to another PR as we discussed